File: implementations/dotnet/Tests/GildedRoseTest.cs
using Xunit;

namespace GildedRose.Tests;

public class GildedRoseTest
{
    [Fact]
    public void IntroImplementationTest()
    {
        Item[] items = new Item[] { new Item("Paisley Pajama Pants", 5, 10) };

        var app = new GildedRose(items);
        int beforeSellIn = app.Items[0].SellIn;
        int beforeQuality = app.Items[0].Quality;

        app.Process();

        // verify normal items, sellIn is decremented by 1
        Assert.Equal(beforeSellIn - 1, app.Items[0].SellIn);

        // verify normal items, quality is decremented by 1
        Assert.Equal(beforeQuality - 1, app.Items[0].Quality);
    }

    [Fact]
    public void IntroBehavioralTest()
    {
        Item[] items = new Item[] { new Item("Paisley Pajama Pants", 5, 10) };

        var app = new GildedRose(items);
        app.Process();

        string actual = app.Items[0].SellIn + ", " + app.Items[0].Quality;
        Assert.Equal("0, 0", actual);
    }
}

File: implementations/java/src/test/java/com/gildedrose/GildedRoseTest.java
package com.gildedrose;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

   @Test
   void introTest() {
        Item[] items = new Item[] { new Item("Paisley Pajama Pants", 5, 10) };

        GildedRose app = new GildedRose(items);
        app.process();

        String actual = app.items[0].sellIn + ", " + app.items[0].quality;
        assertEquals("0, 0", actual);
    }
}
